realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -412,7 +427,8 @@ private void completedAsyncRealmResults(QueryUpdateTask.Result result) {
                         // swap pointer
                         realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
                         // notify callbacks
-                        realmResults.notifyChangeListeners();
+                        realmResults.syncIfNeeded();
```

Comment:
Is this necessary? `callerVersionID.compareTo(result.versionID)` kind of ensure that they have the same version? @nhachicha